tests/test_archive_querier.py

```diff
@@ -290,6 +290,21 @@ def test_paginate_many_records_single_time_bucket(table_maker, querier):
     evaluate_time_based_results(results, 150)
 
 
+def test_paginate_few_records_single_bucket_no_empty_page(table_maker,
+                                                          querier):
+    records = []
+    interval = DatalakeRecord.TIME_BUCKET_SIZE_IN_MS / MAX_RESULTS
+    very_end = DatalakeRecord.TIME_BUCKET_SIZE_IN_MS - 1
+    for start in range(0, very_end, interval):
+        end = start + interval
+        records += create_test_records(start=start, end=end, what='foo')
+    table_maker(records)
+    results = get_multiple_pages(
+        querier.query_by_time,
+        [very_end - 2 * interval + 2, very_end, 'foo'])
+    evaluate_time_based_results(results, 2)
+
```

Comment:
I'm not convinced that this test reliably reproduced the condition we are concerned about. I pasted this test code into master (along with my modified test condition above) and it only seems to fail like 10% of the time.